torch/optim/lr_scheduler.py

```diff
@@ -1,23 +1,40 @@
 import types
 import math
-from torch._six import inf
-from functools import partial, wraps
 import warnings
+from collections import Counter
+from functools import partial, wraps
 from bisect import bisect_right
+from torch._six import inf
 
 from .optimizer import Optimizer
 
 
+EPOCH_DEPRECATION_WARNING = ("The epoch parameter is being deprecated. If epoch "
+                             "is different from None, the closed form is used "
+                             "whenever available instead of the chainable form.")
```

Comment:
Yeah, I'm going to harp on deprecation warnings some more. Put yourself in the shoes of the user: they had some perfectly reasonable, working code on a previous version of PyTorch, and now it's throwing these warnings. What does the warning say? "Epoch is deprecated." OK. Why is it being deprecated? What should I use instead? "If epoch is different from None, the closed form is used whenever available instead of the chainable form." OK.... you didn't answer either of my questions!

Reply:
Thanks again for the feedback. I'll add the "why?" and the "then what?" like so
```
      "The behavior of schedulers has been changed from their closed form to their "                                                                                                                                                     
      "chainable form. As such, the epoch parameter no longer applies, and is being "                                                                                                                                                    
      "deprecated. The new approach is to invoke step without the epoch parameter "                                                                                                                                                      
      "during the iteration. During the deprecation, if epoch is different from None, "                                                                                                                                                  
      " the closed form is used whenever available instead of the chainable form."  
```